### ISSUE TITLE
tail panicked reading file

### ISSUE THREAD

**Issue by adalessandro** [uutils/coreutils#2755]
```
$ head --version
head 0.0.6
$ head /proc/asound/cards
 0 [tlv320aic31xxhi]: tlv320aic31xx-h - tlv320aic31xx-hifi
                      tlv320aic31xx-hifi
$ tail --version
tail 0.0.6
$ RUST_BACKTRACE=full tail /proc/asound/cards
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 22, kind: InvalidInput, message: "Invalid argument" }', src/uu/tail/src/tail.rs:449:44
stack backtrace:
   0:     0xaaaae05805f0 - <unknown>
   1:     0xaaaae05ad738 - <unknown>
   2:     0xaaaae057eaa4 - <unknown>
   3:     0xaaaae0595a90 - <unknown>
   4:     0xaaaae05957bc - <unknown>
   5:     0xaaaae05960c4 - <unknown>
   6:     0xaaaae0580dbc - <unknown>
   7:     0xaaaae0580734 - <unknown>
   8:     0xaaaae0595cbc - <unknown>
   9:     0xaaaae05ac8f4 - <unknown>
  10:     0xaaaae05abbc8 - <unknown>
  11:     0xaaaae0540fec - <unknown>
  12:     0xaaaae053b0a8 - <unknown>
  13:     0xaaaae0540084 - <unknown>
  14:     0xaaaae0531fd0 - <unknown>
  15:     0xaaaae0531fe8 - <unknown>
  16:     0xaaaae05962f0 - <unknown>
  17:     0xaaaae05400e0 - <unknown>
  18:     0xffff9dc0f218 - __libc_start_main
  19:     0xaaaae052b354 - <unknown>
$ 
```

**Comment by hbina** [comment]
Can't reproduce this in `master`.

```shell
hbina@akarin ~/g/uutils (master)> cargo run -- tail /proc/asound/cards
    Finished dev [unoptimized + debuginfo] target(s) in 0.31s
     Running `target/debug/coreutils tail /proc/asound/cards`
 0 [Generic        ]: HDA-Intel - HD-Audio Generic
                      HD-Audio Generic at 0xfe6c8000 irq 96
 1 [Generic_1      ]: HDA-Intel - HD-Audio Generic
                      HD-Audio Generic at 0xfe6c0000 irq 97
hbina@akarin ~/g/uutils (master)> cargo run -- head /proc/asound/cards
    Finished dev [unoptimized + debuginfo] target(s) in 0.32s
     Running `target/debug/coreutils head /proc/asound/cards`
 0 [Generic        ]: HDA-Intel - HD-Audio Generic
                      HD-Audio Generic at 0xfe6c8000 irq 96
 1 [Generic_1      ]: HDA-Intel - HD-Audio Generic
                      HD-Audio Generic at 0xfe6c0000 irq 97
 ```
 
 Or do you want to provide a fix for `0.0.7`? It seems like a lot of things have changed in `tail`.

**Comment by sylvestre** [comment]
@hbina are you planning to propose a PR for this ? :)

**Comment by adalessandro** [comment]
Sorry, perhaps I misunderstood your comments. IIUC, this is fixed in master, isn't it?

**Comment by sylvestre** [comment]
I think @hbina is suggesting that it is more about a special file type not being managed yet :)

**Comment by hbina** [comment]
Sorry for the misunderstanding, AFAIK this is fixed in `master` already so the ticket can be closed unless you intent to provide a fix for `0.0.6` as well? I can't find a proper branch to merge into however, I can't open a PR on a tag.

However, I will admit that I don't fully know why it needs to be like this or if this is sufficient/correct fix. I tried asking around and one answer is that its a driver file and it apparently have some special properties like not being seek-able.

**Comment by tertsdiepraam** [comment]
This issue can indeed be reproduced on 0.0.6 and 0.0.7 and is fixed in 0.0.8, so I think we can close this!
